staging/src/k8s.io/apiserver/pkg/server/options/recommended.go

```diff
@@ -81,6 +97,12 @@ func (o *RecommendedOptions) ApplyTo(config *server.RecommendedConfig) error {
 	if err := o.CoreAPI.ApplyTo(config); err != nil {
 		return err
 	}
+	if o.CoreAPI != nil {
+		// Admission depends on CoreAPI to set SharedInformerFactory and ClientConfig.
+		if err := o.Admission.ApplyTo(&config.Config, config.SharedInformerFactory, config.ClientConfig, scheme, pluginInitializers...); err != nil {
```

Comment:
I wonder, should we return an error if Admission is set, but CoreAPI is nil? I think we should.

Reply:
make sense